Tests/TuistKitTests/Services/GenerateServiceTests.swift

```diff
@@ -109,51 +90,53 @@ final class GenerateCommandTests: TuistUnitTestCase {
     func test_run_withoutPathParameter() throws {
         // Given
         let temporaryPath = try self.temporaryPath()
-        let result = try parser.parse([GenerateCommand.command])
         var generationPath: AbsolutePath?
         generator.generateStub = { path, _ in
             generationPath = path
             return path.appending(component: "Project.xcworkpsace")
         }
 
         // When
-        try subject.run(with: result)
+        try subject.testRun()
 
         // Then
         XCTAssertEqual(generationPath, temporaryPath)
     }
 
     func test_run_withProjectOnlyParameter() throws {
         // Given
-        let arguments = [
-            [GenerateCommand.command, "--project-only"],
-            [GenerateCommand.command],
-        ]
+        let projectOnlyValues = [true, false]
 
         // When
-        try arguments.forEach {
-            let result = try parser.parse($0)
-            try subject.run(with: result)
+        try projectOnlyValues.forEach {
+            try subject.testRun(projectOnly: $0)
         }
 
         // Then
         XCTAssertEqual(generator.generateCalls.map(\.projectOnly), [
             true,
-            false,
+            false
         ])
     }
 
     func test_run_fatalErrors_when_theworkspaceGenerationFails() throws {
         // Given
-        let result = try parser.parse([GenerateCommand.command])
         let error = NSError.test()
         generator.generateStub = { _, _ in
             throw error
         }
 
         // When / Then
-        XCTAssertThrowsError(try subject.run(with: result)) {
+        XCTAssertThrowsError(try subject.testRun()) {
             XCTAssertEqual($0 as NSError, error)
         }
     }
 }
+
+extension GenerateService {
+    func testRun(path: String? = nil,
```

Comment:
> Only new convention to make things easier was to introduce extension for GenerateService called testRun with default function arguments that are equivalent to what is the default outcome of parsing GenerateCommand. This has one downside and that is we will need to make sure that these defaults are synced - so in this regard, I think we probably should come up with a better solution.

Hmm, not sure why we'd like to keep the defaults in sync? 🤔 `GenerateService` should be unit tested in isolation. Relying on the default output from `GenerateCommand` in the unit tests does not feel right.

Reply:
I suppose you are right, `GenerateService` should be decoupled from `GenerateCommand`.

In that case, this approach can actually work just fine without any disadvantages, as far as I can tell.